browser-test/src/applicant/northstar_upsell.test.ts

```diff
@@ -20,32 +20,64 @@ test.describe('Upsell tests', {tag: ['@northstar']}, () => {
       await adminPrograms.expectActiveProgram(programName)
       await logout(page)
     })
+  })
 
+  test('view application submitted page while logged in', async ({
+    page,
+    applicantQuestions,
+  }) => {
     await loginAsTestUser(page)
 
     await enableFeatureFlag(page, 'north_star_applicant_ui')
 
-    await test.step('Setup: submit application', async () => {
+    await test.step('Submit application', async () => {
       await applicantQuestions.clickApplyProgramButton(programName)
       await applicantQuestions.submitFromReviewPage(
         /* northStarEnabled= */ true,
       )
     })
-  })
 
-  test('view application submitted page', async ({page}) => {
     expect(await page.textContent('html')).toContain('Application confirmation')
     expect(await page.textContent('html')).toContain(programName)
 
-    await validateScreenshot(
-      page,
-      'upsell-north-star',
-      /* fullPage= */ true,
-      /* mobileScreenshot= */ true,
-    )
-  })
+    await test.step('Validate screenshot and accessibility', async () => {
+      await validateScreenshot(
+        page,
+        'upsell-north-star',
+        /* fullPage= */ true,
+        /* mobileScreenshot= */ true,
+      )
+    })
 
-  test('passes accessibility checks', async ({page}) => {
     await validateAccessibility(page)
+
+    await test.step('Validate that user can click through without logging in', async () => {
+      await applicantQuestions.clickApplyToAnotherProgramButton()
+      await expect(page.locator('[data-testId="login"]')).toBeHidden()
+    })
+  })
+
+  test('view application submitted page while logged out', async ({
+    page,
+    applicantQuestions,
+  }) => {
+    await enableFeatureFlag(page, 'north_star_applicant_ui')
+
+    await test.step('Submit application', async () => {
+      await applicantQuestions.clickApplyProgramButton(programName)
+      await applicantQuestions.submitFromReviewPage(
+        /* northStarEnabled= */ true,
+      )
+    })
+
+    await test.step('Validate that login dialog is shown when user clicks on apply to another program', async () => {
+      await applicantQuestions.clickApplyToAnotherProgramButton()
+      await expect(page.getByTestId('login')).toContainText(
+        'Create an account or sign in',
+      )
+      await validateScreenshot(page, 'upsell-north-star-login')
+
+      await validateAccessibility(page)
```

Comment:
suggestion (non-blocking): I just tried this out on Firefox and for some reason it doesn't work to tab to the "Log in" button on the modal. I'm able to tab to this button in Chrome, and in Firefox when Voiceover is on, but not with Voiceover off. Would you make an issue to look into this? This could be a Firefox bug and maybe not a problem, but we are able to tab to all the buttons on our current modal so I'm curious what's different. On the plus side, tabbing through the elements never takes you behind the modal which is great!

Reply:
that's weird I can tab through it just fine on firefox...

Reply:
weird! sounds like it's something on my end then.